src/main/java/com/splunk/kafka/connect/SplunkSinkTask.java

```diff
@@ -152,6 +157,66 @@ private void handleRaw(final Collection<SinkRecord> records) {
         }
     }
 
+    private void handleRecordsWithHeader(final Collection<SinkRecord> records) {
+        HashMap<String, ArrayList<SinkRecord>> recordsWithSameHeaders = new HashMap<>();
+
+        for (SinkRecord record : records) {
+            String key = headerId(record);
+            if (!recordsWithSameHeaders.containsKey(key)) {
+                ArrayList<SinkRecord> recordList = new ArrayList<SinkRecord>();
+                recordsWithSameHeaders.put(key, recordList);
+            }
+            ArrayList<SinkRecord> recordList = recordsWithSameHeaders.get(key);
+            recordList.add(record);
+            recordsWithSameHeaders.put(key, recordList);
+        }
+
+        int index = 0;
+        Iterator<Map.Entry<String, ArrayList<SinkRecord>>> itr = recordsWithSameHeaders.entrySet().iterator();
+        while(itr.hasNext()) {
+            Map.Entry set = itr.next();
+            String splunkSinkRecordKey = (String)set.getKey();
+            ArrayList<SinkRecord> recordArrayList = (ArrayList)set.getValue();
+            EventBatch batch = createRawHeaderEventBatch(splunkSinkRecordKey);
+            sendEvents(recordArrayList, batch);
+            index++;
+        }
+        log.debug("{} records have been bucketed in to {} batches",records.size(), index);
```

Comment:
look like the formatting of this line is not good ? after ",", there shall be space ?

Reply:
we don't need maintain a `index` as `recordsWithSameHeaders.size()` can serve the same purpose ?